### PR TITLE
Rich text pub field type

### PR DESCRIPTION
**Pull request by allisonking**
## Issue(s) Resolved
Closes https://github.com/pubpub/platform/issues/720

And some progress on https://github.com/pubpub/platform/issues/721

## High-level Explanation of PR
This makes two enum changes to add rich text to both `CoreSchemaType` and `InputComponent`. Many of the changes made here were what follows from adding these enum types (we already had good type checking in place that makes sure we handle cases for every enum!). 

Now you can add a pub field of "Rich Text" type, and you can add that to forms or pub types. You can save a form configuration for the rich text field, but we don't display it anywhere yet (that's the rest of https://github.com/pubpub/platform/issues/721, and is blocked by https://github.com/pubpub/platform/issues/722)

There is also my guess at validating a prose mirror field, though I don't have a way to test it out yet and am guessing it probably won't just work as-is!

## Test Plan
* Add a new pub field of RichText type
* You should be able to add it to a pub type, but creating a pub of this pub type won't render the rich text field yet
* You should be able to add it to a form and configure a "label" and "help text" but this won't render the rich text field in the fill form page yet

## Screenshots (if applicable)

image
image

### CODE REVIEW

File: core/actions/_lib/validateFields.ts [hunk 1]
```diff
@@ -28,8 +30,21 @@ export const validatePubValuesBySchemaName = ({
...
 		}
-		const jsonSchema = getJsonSchemaByCoreSchemaType(field.schemaName);
-		const result = Value.Check(jsonSchema, value);
+
+		let result = false;
+		// Rich text fields are a special case where we use prosemirror to validate
+		// as opposed to typebox
+		if (field.schemaName === CoreSchemaType.RichText) {
+			try {
+				(value as Node).check();
```
Comment (allisonking) [inline core/actions/_lib/validateFields.ts:39]: I really don't know if this will work yet...

Comment (tefkah) [reply core/actions/_lib/validateFields.ts:39]: I think, as the `value` will be a `JSON` value, we will first need to call `nodeFromJSON` on the `schema` @isTravis defines here: https://github.com/pubpub/platform/blob/tr/context-editor/packages/context-editor/src/schemas/index.ts

Not sure how you would do this in this PR though 😅. Maybe you extract the prosemirror `nodeFromJSON` + `Node.check` validation to a different file/function, and create a stub schema there for now with a comment indicating we will import it from the editor?

File: core/package.json [hunk 2]
```diff
@@ -91,6 +91,7 @@
 		"openapi3-ts": "^4.1.2",
 		"pg": "^8.11.3",
 		"prosemirror-markdown": "^1.12.0",
```
Comment (tefkah) [inline core/package.json:93]: why did we already have `prosemirror-markdown` ??

Comment (allisonking) [reply core/package.json:93]: hmm looks like for the `pushToV6` action! 

https://github.com/pubpub/platform/blob/1f3217c7319d29356a160726076fc1a1e6a87778/core/actions/pushToV6/run.ts#L136-L138

File: core/lib/__tests__/contextEditor.test.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	],
+	type: "doc",
+};
+
+test.each([
+	{ value: "", expected: false },
+	{ value: 1, expected: false },
+	{ value: {}, expected: false },
+	{ value: JSON_EXAMPLE, expected: true },
+])("validateAgainstContextEditorSchema", ({ value, expected }) => {
```
[the remark this answers is not in the document]
Comment (allisonking) [reply core/lib/__tests__/contextEditor.test.ts:25]: never mind—let me make a better test!

File: core/package.json (hunk 2 above)
Comment (3mcd) [reply core/package.json:93]: yea i used it to convert markdown to a prosemirror doc before creating the pub in v6

Comment (tefkah) [reply core/package.json:93]: ahh that makes sense

**Pull request merged**